evalml/tests/component_tests/test_ordinal_encoder.py

```diff
@@ -0,0 +1,1019 @@
+        encoder.fit(X)
+
+
+def test_ordinal_encoder_features_to_encode_no_col_names():
+    X = pd.DataFrame([["b", 0], ["a", 1], ["b", 1]])
+    X.ww.init(
+        logical_types={
+            0: Ordinal(order=["b", "a"]),
+            1: Ordinal(order=[0, 1]),
+        },
+    )
+    encoder = OrdinalEncoder(top_n=5, features_to_encode=[0])
+    encoder.fit(X)
+    X_t = encoder.transform(X)
+    expected_col_names = set([1, "0_ordinally_encoded"])
+    col_names = set(X_t.columns)
+    assert col_names == expected_col_names
+    assert [X_t[col].dtype == "uint8" for col in X_t]
+
+
+def test_ordinal_encoder_top_n_categories_always_the_same():
+    df = pd.DataFrame(
+        {
+            "categories": ["cat_1"] * 5
+            + ["cat_2"] * 4
+            + ["cat_3"] * 3
+            + ["cat_4"] * 3
+            + ["cat_5"] * 3,
+            "numbers": range(18),
+        },
+    )
+    df.ww.init(
+        logical_types={
+            "categories": Ordinal(order=["cat_1", "cat_2", "cat_3", "cat_4", "cat_5"]),
+        },
+    )
+
+    def check_df_equality(random_seed):
+        ordinal_encoder = OrdinalEncoder(
+            top_n=4,
+            random_seed=random_seed,
+            handle_unknown="use_encoded_value",
+            unknown_value=-1,
+        )
+
+        df1 = ordinal_encoder.fit_transform(df)
+        df2 = ordinal_encoder.fit_transform(df)
+        pd.testing.assert_frame_equal(df1, df2)
+
+    check_df_equality(5)
+    check_df_equality(get_random_seed(5))
+
+
+def test_ordinal_encoder_output_doubles():
+    X = pd.DataFrame(
+        {
+            "bool": [bool(i % 2) for i in range(100)],
+            "categorical": ["dog"] * 20 + ["cat"] * 40 + ["fish"] * 40,
+            "integers": [i for i in range(100)],
+            "doubles": [i * 1.0 for i in range(100)],
+        },
+    )
+    X.ww.init(
+        logical_types={
+            "categorical": Ordinal(order=["dog", "cat", "fish"]),
+        },
+    )
+    y = pd.Series([i % 2 for i in range(100)])
+    y.ww.init()
+    ordinal_encoder = OrdinalEncoder()
+    output = ordinal_encoder.fit_transform(X, y)
+    for name, types in output.ww.types["Logical Type"].items():
+        if name == "integers":
+            assert str(types) == "Integer"
+        elif name == "bool":
+            assert str(types) == "Boolean"
+        else:
+            assert str(types) == "Double"
+    assert len(output.columns) == len(X.columns)
+
+
+@pytest.mark.parametrize("data_type", ["list", "np", "pd_no_index", "pd_index", "ww"])
+def test_data_types(data_type):
+    if data_type == "list":
+        X = [["a"], ["b"], ["c"]] * 5
+    elif data_type == "np":
+        X = np.array([["a"], ["b"], ["c"]] * 5)
+    elif data_type == "pd_no_index":
+        X = pd.DataFrame(["a", "b", "c"] * 5)
+    elif data_type == "pd_index":
+        # --> doing int 0 here might defeat the purpose of the no index one?
+        X = pd.DataFrame(["a", "b", "c"] * 5, columns=[0])
```

Comment:
Not totally sure what this case was testing when `columns=["0"]`, but I think me setting the index to be an integer means it's essentially the same as the `pd_no_index` case, but given that this test can only actually perform ordinal encoding in the `ww` case (because Ordinal will never be inferred and so won't be used unless woodwork is initialized), I'm not sure this test is as useful as it is for the one-hot encoder. Maybe I should just remove it?

Reply:
We do still want to make sure none of our components break when passed any standard data type, so we should still have a test that throws them at the encoder making sure it doesn't fall over. I think you're right in that we don't need to differentiate about the pandas index, so we can probably condense this to just test list, np, pd, and ww instead, and make sure nothing happens in any case except woodwork.

(And in that case I think we can remove the numpy test on line 744 since we're doing it here instead)

Reply:
Done!